### PR TITLE
DOC/MAINT: optimize: fix some typos

### PR DESCRIPTION
**Pull request by musvaage**
@lucascolley

dumped directory strings exposed a few typos

the sole C language files modified are in the tnc subdirectory

check that the replacement string in this file is correct

```
$ grep -nr neigbouds scipy/scipy/optimize
scipy/scipy/optimize/_shgo_lib/_complex.py:1049:                        pass  # NOTE: Not all neigbouds are in initial pool
$ 
```

### CODE REVIEW

**Comment by musvaage** [comment]
> dumped directory strings

misc and odr

in the former no indicated string corrections

the latter's odrpack subdirectory's  files contain a few typos; Makefile has an unnecessary string `d_lpkbls.o`

File: scipy/optimize/_trustregion_exact.py [hunk 1]
```diff
@@ -246,7 +246,7 @@ def __init__(self, x, fun, jac, hess, hessp=None,
         self.hess_fro = norm(self.hess, 'fro')
 
         # A constant such that for vectors smaller than that
-        # backward substituition is not reliable. It was stabilished
+        # backward substitution is not reliable. It was stabilised
```
Comment (rkern) [inline scipy/optimize/_trustregion_exact.py:249]:
```suggestion
        # backward substitution is not reliable. It was estabilished
```

**Review by lucascolley — APPROVED** [review]
thanks!

**Comment by musvaage** [comment]
in the odrpack directory I'm neither going to implement the typo fixes nor remove the errant string from the Makefile

**Comment by musvaage** [comment]
_this is only for my own records_

`https://github.com/xiaoyeli/superlu/pull/159`

typo fixes were merged in SuperLU; .html files not regenerated which as I'm informed may happen prior to a new release

@dschmitz89 and others have committed to the bundled code

**Pull request merged**